src/sparseml/pytorch/recipe_template/main.py

```diff
@@ -0,0 +1,376 @@
+    MFACPruningModifier,
+    MovementPruningModifier,
+    OBSPruningModifier,
+    QuantizationModifier,
+)
+from sparseml.pytorch.utils import get_prunable_layers
+from sparseml.sparsification import ModifierYAMLBuilder, RecipeYAMLBuilder
+
+
+__all__ = [
+    "recipe_template",
+]
+
+
+def recipe_template(
+    pruning: Optional[str] = None,
+    quantization: bool = False,
+    lr_func: str = "linear",
+    mask_type: str = "unstructured",
+    global_sparsity: bool = False,
+    target: Optional[str] = None,
+    model: Union[str, Module, None] = None,
+    file_name: Optional[str] = None,
+) -> str:
+    """
+    Returns a valid yaml or md recipe based on specified arguments
+
+    :pruning: optional pruning algorithm to use in the recipe, can be any of the
+    following,
+        `true` (represents Magnitude/Global-Magnitude pruning according to
+        global_sparsity), `false` (No pruning), `acdc`, `mfac`, `movement`, `obs` or
+        `constant`. Defaults to None
+    :quantization: True if quantization needs to be applied else False. Defaults to
+        False
+    :lr_func: the learning rate schedule function. Defaults to `linear`
+    :mask_type: the mask_type to use for pruning. Defaults to `unstructured`
+    :global_sparsity: if set to True then apply sparsity globally, defaults to
+        False
+    :target: the target hardware, can be set to `vnni` or `tensorrt`. Defaults to
+        None
+    :model: an instantiated PyTorch Module, or the local path to a torch.jit loadable
+        *.pt file, if supplied then the recipe is built according to this architecture
+    :file_name: an optional filename to save this recipe to. If specified the extension
+        is used to determine if file should be written in markdown or yaml syntax. If
+        not specified recipe is not written to a file
+    """
+
+    if isinstance(model, str):
+        # load model file to in memory Module using torch.jit
+        model = torch.jit.load(model)
+
+    mask_type: str = _validate_mask_type(mask_type=mask_type, target=target)
+    pruning: str = _validate_pruning(pruning=pruning, quantization=quantization)
+    recipe: str = _build_recipe_template(
+        pruning=pruning,
+        quantization=quantization,
+        lr_func=lr_func,
+        mask_type=mask_type,
+        global_sparsity=global_sparsity,
+        target=target,
+        model=model,
+    )
+
+    if file_name is not None:
+        if Path(file_name).suffix == ".md":
+            recipe = _add_description(recipe=recipe)
+
+        _write_recipe_to_file(file_name=file_name, recipe=recipe)
+    return recipe
+
+
+def _validate_pruning(pruning: Optional[str] = None, quantization: bool = False) -> str:
+    # normalize pruning algo name
+    pruning = (pruning or "").lower().replace("/", "")
+
+    if pruning == "true":
+        pruning = "magnitude"
+    elif pruning == "false":
+        pruning = "constant" if quantization else ""
+    return pruning
```

Comment:
Still wondering about "true"/"false" in strings from cli/yaml PoV, but unsure if this is an issue.

Reply:
Let's make a note of this and keep it as is for now?